include/librealsense2/dds/dds-device.cpp

```diff
@@ -167,6 +167,61 @@ dds_device::foreach_motion_profile( size_t sensor_index,
     return msg->second.num_of_profiles;
 }
 
+void dds_device::sensor_open( size_t sensor_index, const std::vector< rs2_video_stream > & profiles )
+{
+    using namespace librealsense::dds::topics;
+
+    assert( profiles.size() < device::control::MAX_OPEN_PROFILES );
+
+    device::control::sensor_open_msg open_msg;
+    open_msg.message_id = _impl->_message_counter++;
+    open_msg.sensor_uid = static_cast<uint16_t>( sensor_index );
+    for ( size_t i = 0; i < profiles.size(); ++i )
+    {
+        open_msg.profiles[i].framerate = profiles[i].fps;
+        open_msg.profiles[i].format = profiles[i].fmt;
+        open_msg.profiles[i].type = profiles[i].type;
+        open_msg.profiles[i].width = profiles[i].width;
+        open_msg.profiles[i].height = profiles[i].height;
+    }
+
+    raw::device::control raw_msg;
+    device::control::construct_raw_message( device::control::control_type::SENSOR_OPEN,
+                                            open_msg,
+                                            raw_msg );
+
+    if ( _impl->write_control_message( &raw_msg ) )
```

Comment:
Not `DDS_API_CALL`?

Reply:
No, this is `dds_device::impl` function. Inside it I call `DDS_API_CALL`